### ISSUE TITLE
Cannot use complex expressions like `hash['foo'] += 1` with hyper-spec

### ISSUE THREAD

**Issue by catmando** [hyperstack-org/hyperstack#127]
This is due to an issue in the current unparser gem: https://github.com/mbj/unparser/issues/117

You may get an error like this when running a spec:

```text
(string):1:20: error: unexpected token tOP_ASGN
(string):1: hash.[]("foo") += 1
(string):1: 
```

The problem is the unparser incorrectly generates `hash.[]("foo") += 1` instead of `hash['foo'] += 1`.

The good news its pretty easy to find such expressions and replace them with something like 

`hash["foo"] = hash["foo"] + 1`

**Comment by mbj** [comment]
You need to pass the [modern AST](https://github.com/whitequark/parser/#usage) format to unparser.

**Comment by catmando** [comment]
works with parser 3.0 and all settings on per mbj's comment.

**Comment by catmando** [comment]
closed in 1.0.alpha1.7
